### PR TITLE
add user guides for data and remote repos

### PR DESCRIPTION
**Pull request by madhur-tandon**
Closes https://github.com/iterative/mlem.ai/issues/70

I just restored the pages + made sure they follow the new convention after renaming. Not sure if anything more is to be done here?

CC: @aguschin

### CODE REVIEW

File: content/docs/user-guide/data.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+To use data within the MLEM context, so that it is compatible with other operations
+such as [mlem apply](/doc/command-reference/apply), one needs to save it as a
+[MLEM Object](/doc/user-guide/basic-concepts#mlem-objects).
+
+## Saving data with MLEM
+
+One can use MLEM to save existing data using the [mlem.api.save()](/doc/api-reference/save) API.
+
+As an example, we load the well-known IRIS dataset with `sklearn`, and then save parts of it
+with MLEM locally.
```
Comment (omesser) [inline content/docs/user-guide/data.md:38]:
```suggestion
locally using MLEM.
```

File: content/docs/user-guide/data.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+```py
+# prepare.py
```
Comment (omesser) [inline content/docs/user-guide/data.md:41]: 1. This comment is a bit weird imo. marking the file of which this is the contents. Suggest to move this "comment" out of the fenced block and instead have this above the block:
> Consider this `prepare.py` file:

2. I would add a shebang instead actually - `#!/usr/bin/env python` (followed by 2 empty lines as is tradition)

File: content/docs/user-guide/data.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      - ''
+    type: dataframe
+  format: csv
+  type: pandas
+requirements:
+  - module: pandas
+    version: 1.4.2
+```
+
+</details>
```
Comment (omesser) [inline content/docs/user-guide/data.md:121]: I think we should remove the details block here. the yaml is not very long. Also the explanation in lines 123-128 will not make sense if the entire snippet is hidden, and showing the `mlem` file's contents is too important to completely fold into "details" (IMO)

File: content/docs/user-guide/remote-projects.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+# Working with projects and remote objects
+
+<details>
```
Comment (omesser) [inline content/docs/user-guide/remote-projects.md:3]: This detail block needs to be removed IMO. this page doesn't use the `dvc` branch of the [example repo](https://github.com/iterative/example-mlem-get-started) at all

**Review by omesser — DISMISSED** [review]
Thanks @madhur-tandon !
Did a quick review, please take a look

File: content/docs/user-guide/data.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
+# Working with data
+
+<details>
+
+### ⚙️ Expand for setup instructions
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:5]: CC: @aguschin, should we do this?

File: content/docs/user-guide/data.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+</details>
+
+To use data within the MLEM context, so that it is compatible with other operations
+such as [mlem apply](/doc/command-reference/apply), one needs to save it as a
+[MLEM Object](/doc/user-guide/basic-concepts#mlem-objects).
+
+## Saving data with MLEM
+
+One can use MLEM to save existing data using the [mlem.api.save()](/doc/api-reference/save) API.
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:35]: done

File: content/docs/user-guide/data.md (hunk 1 above)
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:38]: done

File: content/docs/user-guide/data.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    train_data, test_data = train_test_split(data, random_state=42)
+    save(train_data, "train.csv")
+    save(test_data.drop("target", axis=1), "test_x.csv")
+    save(test_data[["target"]], "test_y.csv")
+
+if __name__ == "__main__":
+    main()
+```
+
+One can see what files were produced by executing the above script:
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:58]: done

File: content/docs/user-guide/data.md (hunk 2 above)
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:41]: done

File: content/docs/user-guide/data.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+├── test_x.csv
+├── test_x.csv.mlem
+├── test_y.csv
+├── test_y.csv.mlem
+├── train.csv
+└── train.csv.mlem
+```
+
+Clearly, every DataFrame was saved along with some metadata
+about it. One can of course check the contents of these files.
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:73]: done

File: content/docs/user-guide/data.md (hunk 3 above)
Comment (madhur-tandon) [reply content/docs/user-guide/data.md:121]: done

File: content/docs/user-guide/remote-projects.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+model = load(
+    "rf",
+    project="https://github.com/iterative/example-mlem-get-started",
+    rev="simple"
+)
+```
+
+## Cloning objects
+
+One can also download the object to local disk so as to use it later
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:63]: done

File: content/docs/user-guide/remote-projects.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+using the `clone` command.
+
+```cli
+$ mlem clone https://github.com/iterative/example-mlem-get-started/rf ml_model
+```
+
+The other way to do this is to run
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:70]: done

File: content/docs/user-guide/remote-projects.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  - test_x.csv
+  - test_y.csv
+  - train.csv
+Models:
+  - rf
+```
+
+Here, the project being considered is the one we're currently in.
+
+But one can list objects inside a remote Git repository as well, without
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:35]: done

File: content/docs/user-guide/remote-projects.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+</details>
+
+## Cloud remotes
+
+MLEM can also be used with any cloud/remote supported by
+[fsspec](https://filesystem-spec.readthedocs.io/en/latest/api.html#built-in-implementations),
+e.g. s3. This is useful in scenarios where one may want to store objects in some remote location
+without the need to version them.
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:98]: done

File: content/docs/user-guide/remote-projects.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+$ git push
+```
+
+</details>
+
+## Cloud remotes
+
+MLEM can also be used with any cloud/remote supported by
+[fsspec](https://filesystem-spec.readthedocs.io/en/latest/api.html#built-in-implementations),
+e.g. s3. This is useful in scenarios where one may want to store objects in some remote location
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:97]: done

File: content/docs/user-guide/remote-projects.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+```cli
+$ mlem init s3://example-mlem-get-started
+$ mlem clone rf s3://example-mlem-get-started/rf
+⏳️ Loading meta from .mlem/model/rf.mlem
+🐏 Cloning .mlem/model/rf.mlem
+💾 Saving model to s3://example-mlem-get-started/.mlem/model/rf.mlem
+```
+
+This model can now be loaded via API or can be used in CLI commands just like if it
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:111]: done

File: content/docs/user-guide/remote-projects.md (hunk 4 above)
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:3]: cc: @aguschin why did we consider this in the original content?

File: content/docs/user-guide/remote-projects.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+The other way to do this is to run
+
+```cli
+$ mlem clone rf --project https://github.com/iterative/example-mlem-get-started --rev main ml_model
+```
+
+<details>
+
+### 💡 Expand to use your own repo
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply content/docs/user-guide/remote-projects.md:80]: CC: @aguschin

[the remark this answers is not in the document]
Comment (aguschin) [reply content/docs/user-guide/remote-projects.md]: Yep, that's not needed! Removed.

File: content/docs/user-guide/data.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (aguschin) [reply content/docs/user-guide/data.md:5]: Kept `pip install` and venv. Removed unnecessary part.

**Review by aguschin — APPROVED** [review]
I'm merging this. We will address improvements and feedback for these pages later, together with updating other pages I think.

**Comment by jorgeorpinel** [comment]
Since I didn't get to review this before, I'm opening a follow-up with some level of rewriting + some high-level feedback that applies to these but probably all guides. See https://github.com/iterative/mlem.ai/pull/113

Thanks

**Pull request merged**